### ISSUE TITLE
add new modules for each possible monad instance

### ISSUE THREAD

**Issue by polymorphicengine** [tidalcycles/Tidal#1004]
I just stumbeled upon the QualifiedDo GHC extension and found that it provides a functionality that I was really missing/wanting:
choosing between monad instances when using the do-notation. Since `Pattern` has many natural monad instances this comes in very handy

to use this we have to add a module for each monad instance that implement `(>>=)`, which are then supposed to be imported qualified

for example:

```
module Sound.Tidal.Monad.Outer where

import Sound.Tidal.Pattern

(>>=) :: Pattern a -> (a -> Pattern b) -> Pattern b
p >>= f = outerJoin (f <$> p)
```

and 

```
module Sound.Tidal.Monad.Inner where

import Sound.Tidal.Pattern

(>>=) :: Pattern a -> (a -> Pattern b) -> Pattern b
p >>= f = innerJoin (f <$> p)
```

etc.

then in `Sound.Tidal.Context`

```
import qualified Sound.Tidal.Monad.Outer as O
import qualified Sound.Tidal.Monad.Inner as I
```
etc.

then one can write code like

```
O.do
x <- run 8
y <- "[0 | 1]*8"
return $ x + y
```

we could also discuss which monad instance should be the default one.. right now it uses `unwrap` which combines the structures of the inner and outer patterns - i think the default should probably be taking the structure of the outer

**Comment by yaxu** [comment]
ah great!

**Comment by sss-create** [comment]
What's the status on this?

**Comment by polymorphicengine** [comment]
there's more discussion about this here:
https://club.tidalcycles.org/t/instance-monad-pattern-should-use-squeezejoin-instead-of-unwrap/5560/8

i'm still a bit undecided if it's actually gonna be useful..

**Comment by yaxu** [comment]
Yep this looks nice but it would likely only be really useful if you can qualify the bind for each computation inside the monad, if that makes sense.
